offchain-modules/src/packages/handlers/eos.ts

```diff
@@ -0,0 +1,212 @@
+import { logger } from '../utils/logger';
+import { asyncSleep } from '../utils';
+import { EosDb } from '@force-bridge/db/eos';
+import { EosChain } from '@force-bridge/xchain/eos/eosChain';
+import { EosUnlock, EosUnlockStatus } from '@force-bridge/db/entity/EosUnlock';
+import { TransactResult } from 'eosjs/dist/eosjs-api-interfaces';
+import { PushTransactionArgs } from 'eosjs/dist/eosjs-rpc-interfaces';
+import { EosConfig } from '@force-bridge/config';
+import { parseAssetAmount } from '@force-bridge/xchain/eos/utils';
+import { JsSignatureProvider } from 'eosjs/dist/eosjs-jssig';
+import { ChainType } from '@force-bridge/ckb/model/asset';
+
+const EosDecimal = 4;
+const EosTokenAccount = 'eosio.token';
+const EosTokenTransferActionName = 'transfer';
+
+export class EosLockEvent {
+  TxHash: string;
+  BlockNumber: number;
+  AccountActionSeq: number;
+  GlobalActionSeq: number;
+  Asset: string;
+  From: string;
+  To: string;
+  Amount: string;
+  Memo: string;
+}
+
+export class EosHandler {
+  private db: EosDb;
+  private config: EosConfig;
+  private chain: EosChain;
+  constructor(db: EosDb, config: EosConfig) {
+    this.db = db;
+    this.config = config;
+    this.chain = new EosChain(this.config.rpcUrl, new JsSignatureProvider(this.config.privateKeys));
+  }
+
+  async getUnlockRecords(status: EosUnlockStatus): Promise<EosUnlock[]> {
+    return this.db.getEosUnlockRecordsToUnlock(status);
+  }
+
+  async sendUnlockTx(record: EosUnlock): Promise<TransactResult | PushTransactionArgs> {
+    return this.chain.transfer(
+      this.config.bridgerAccount,
+      record.recipientAddress,
+      this.config.bridgerAccountPermission,
+      `${record.amount} ${record.asset}`,
+      '',
+      EosTokenAccount,
+    );
+  }
+
+  async watchLockEvents() {
+    //check chain id
+    const curBlockInfo = await this.chain.getCurrentBlockInfo();
+    if (curBlockInfo.chain_id != this.config.chainId) {
+      logger.error(`Eos chainId:${curBlockInfo.chain_id} doesn't match with:${this.config.chainId}`);
+      return;
+    }
+
+    let latestActionSeq = await this.db.getLastedAccountActionSeq();
+    if (latestActionSeq < this.config.latestAccountActionSeq) {
+      latestActionSeq = this.config.latestAccountActionSeq;
+    }
+
+    let pos = 0;
+    let forward = false;
+    const offset = 20;
+    while (true) {
+      if (pos < 0) {
```

Comment:
It's a little bit hard to understand the loops and branches. May need to refactor.

Reply:
This loop is complex become of the result of history_get_actions return result ordered by desc, and the index of action was not stable